http-server/src/main/java/xtdb/query/QueryMap.java

```diff
@@ -0,0 +1,11 @@
+package xtdb.query;
+
+import clojure.lang.Keyword;
+
+import java.time.Duration;
+import java.time.ZoneId;
+import java.util.Map;
+
+public record QueryMap(Query query, Map<Keyword, Object> args, Basis basis, TransactionKey afterTx, Duration txTimeout,
```

Comment:
similarly to `Basis`, I suspect this is a candidate Java API class.

nit: naming: `QueryRequest`?

Reply:
also, `boolean` rather than `Boolean` if it can never be null

Reply:
and, if it is to become the Java API, we should expect `Map<String, Object> args` here, rather than making the Java folks write Clojure keywords

Reply:
Still yet to update the types of `args` here but I have moved it - shall update that on monday.

Reply:
Was just simple String replacement.